target/cheri-common/op_helper_cheri_common.c

```diff
@@ -265,6 +316,27 @@ target_ulong CHERI_HELPER_IMPL(cjalr(CPUArchState *env, uint32_t cd,
 #ifdef CONFIG_DEBUG_TCG
     env->_pc_is_current = true;
 #endif
+#elif defined(TARGET_AARCH64)
+    // TODO I want to move this out of common code.
+    bool system_changed = ((env->pc.cr_perms ^ next_pcc.cr_perms) & CAP_ACCESS_SYS_REGS) != 0;
+    bool executive_changed = ((env->pc.cr_perms ^ next_pcc.cr_perms) & CAP_PERM_EXECUTIVE) != 0;
+    bool c64_changed = (next_pcc._cr_cursor & 1) != ((env->pstate & PSTATE_C64) != 0);
+
+    next_pcc._cr_cursor &= ~1;
+    env->pc = next_pcc;
+    if(executive_changed) {
+        // LETODO: If in executive and target is not, check we have the CJALR_CAN_BRANCH_RESTRICTED flag
+        // LETODO: Other direction is just always allowed?
+        assert(0);
+    }
+
+    // Switch C64. Branches will have already ended a translation block.
+    if(c64_changed)
+        env->pstate ^= PSTATE_C64;
+
+    // Rebuild cached target-specific cheriflags if anything changed
+    if(system_changed | executive_changed | c64_changed)
```

Comment:
```suggestion
    if (system_changed || executive_changed || c64_changed)
```
logical or for `bool` variables

Reply:
Fixed